src/main/java/org/folio/linked/data/service/resource/ResourceMarcServiceImpl.java

```diff
@@ -230,8 +245,43 @@ private boolean isMonograph(String leader) {
     return isLanguageMaterial(leader.charAt(6)) && isMonographicComponentPartOrItem(leader.charAt(7));
   }
 
+  private Optional<ResponseEntity<Record>> getRecord(String inventoryId) {
+    try {
+      return Optional.of(srsClient.getFormattedSourceStorageInstanceRecordById(inventoryId));
+    } catch (FeignException.NotFound e) {
+      return Optional.empty();
+    }
+  }
+
+  private Optional<org.folio.ld.dictionary.model.Resource> getResource(String inventoryId) {
+    return getRecord(inventoryId)
+      .map(HttpEntity::getBody)
+      .map(Record::getParsedRecord)
+      .map(ParsedRecord::getContent)
+      .map(this::toJsonString)
+      .flatMap(marcBib2ldMapper::fromMarcJson);
+  }
+
+  private Long save(org.folio.ld.dictionary.model.Resource modelResource) {
+    var id = modelResource.getId();
+    if (resourceRepo.existsById(id)) {
+      throw new AlreadyExistsException(format("Another resource with ID %s already exists in the graph", id));
+    }
+
+    var srsId = modelResource.getFolioMetadata().getSrsId();
+    if (folioMetadataRepository.existsBySrsId(srsId)) {
+      throw new AlreadyExistsException(format("MARC record having srsID %s is already imported to data graph", srsId));
+    }
+
```

Comment:
Hi @askhat-abishev, I do not think anything will get logged when the API call is made. Can you please ensure that log messages are added?